src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -281,11 +282,91 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto actionPaletteItem{ filteredCommand.Item().try_as<winrt::TerminalApp::ActionPaletteItem>() })
             {
-                PreviewAction.raise(*this, actionPaletteItem.Command());
+                const auto& cmd = actionPaletteItem.Command();
+                PreviewAction.raise(*this, cmd);
+
+                const auto description{ cmd.Description() };
+
+                if (const auto& selected{ SelectedItem() })
+                {
+                    selected.SetValue(Automation::AutomationProperties::FullDescriptionProperty(), winrt::box_value(description));
+                }
+
+                if (!description.empty())
+                {
+                    _openTooltip(cmd);
+                }
+                else
+                {
+                    // If there's no description, then just close the tooltip.
+                    _descriptionsView().Visibility(Visibility::Collapsed);
+                    _descriptionsBackdrop().Visibility(Visibility::Collapsed);
+                    _recalculateTopMargin();
+                }
             }
         }
     }
 
+    winrt::fire_and_forget SuggestionsControl::_openTooltip(Command cmd)
+    {
+        const auto description{ cmd.Description() };
+        if (description.empty())
+        {
+            co_return;
+        }
+
+        // If it's already open, then just re-target it and update
+        // the content immediately. Otherwise:
+        // wait a bit before opening it.
+        if (_descriptionsView().Visibility() != Visibility::Visible)
+        {
+            co_await winrt::resume_after(200ms);
+            co_await wil::resume_foreground(Dispatcher());
+        }
+
+        // Build the contents of the "tooltip" based on the description
+        //
+        // First, the title. This is just the name of the command.
+        _descriptionTitle().Inlines().Clear();
+        Documents::Run titleRun;
+        titleRun.Text(cmd.Name());
+        _descriptionTitle().Inlines().Append(titleRun);
+
+        // Now fill up the "subtitle" part of the "tooltip" with the actual
+        // description itself.
+        _descriptionComment().Inlines().Clear();
+
+        // First, replace all "\r\n" with "\n"
+        std::wstring filtered = description.c_str();
+        std::wstring::size_type pos = 0;
+        while ((pos = filtered.find(L"\r\n", pos)) != std::wstring::npos)
+        {
+            filtered.erase(pos, 1);
+        }
+
+        // Split the filtered description on '\n`
+        const auto lines = ::Microsoft::Console::Utils::SplitString(filtered.c_str(), L'\n');
+        // build a Run + LineBreak, and add them to the text block
+        for (const auto& line : lines)
+        {
+            if (line.empty())
+            {
+                continue;
+            }
+            Documents::Run textRun;
+            textRun.Text(winrt::hstring{ line });
```

Comment:
I don't think you need to explicitly wrap `line` in an `hstring`.
This is only necessary for the `til` helpers and macros that we built ourselves, because we don't properly use `winrt::param::hstring`. But otherwise?